lldb/source/Target/StackFrame.cpp

```diff
@@ -475,6 +476,32 @@ VariableList *StackFrame::GetVariableList(bool get_file_globals,
   return m_variable_list_sp.get();
 }
 
+// BEGIN SWIFT
+// Implement LanguageCPlusPlus::GetParentNameIfClosure and upstream this.
+// rdar://152321823
+
+/// If `sc` represents a "closure-like" function according to `lang`, and
+/// `missing_var_name` can be found in a parent context, create a diagnostic
+/// explaining that this variable is available but not captured by the closure.
+static std::string
+GetVariableNotCapturedDiagnostic(SymbolContext &sc, SourceLanguage lang,
+                                 ConstString missing_var_name) {
+  Language *lang_plugin = Language::FindPlugin(lang.AsLanguageType());
+  if (lang_plugin == nullptr)
+    return "";
+  Function *parent_func =
+      lang_plugin->FindParentOfClosureWithVariable(missing_var_name, sc);
+  if (!parent_func)
+    return "";
+  return llvm::formatv("Current frame is a closure.\nA variable named '{0}' "
```

Comment:
Why is the text duplicated here?

Reply:
>The second commit implements the v diagnostics in a language agnostic way. It is what should be upstreamed to LLVM once we can support this in C++. I punted on doing this because C++ mangling is a bit <redacted> to work with.

I see.